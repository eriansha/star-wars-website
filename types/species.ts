export interface Species {
  id: string
  classification: string
  name: string
}
