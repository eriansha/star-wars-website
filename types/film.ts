import { Person } from '@/types/person'

export interface Film {
  id?: string
  title: string
  episodeID: number
  openingCrawl: string
  director: string
  producers: string[]
  characterConnection?: FilmCharacterConnection
}

export interface FilmCharacterConnection {
  characters: Person[]
}

export interface FilmEdge {
  node: Film
  cursor: string
}
