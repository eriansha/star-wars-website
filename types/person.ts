import { Planet } from '@/types/planet'
import { Species } from '@/types/species'

export interface Person {
  id: string
  name: string
  birthYear?: string
  eyeColor?: string
  gender?: string
  hairColor?: string
  height?: number
  masss?: number
  skinColor?: string
  homeWorld?: Planet
  species?: Species
}
