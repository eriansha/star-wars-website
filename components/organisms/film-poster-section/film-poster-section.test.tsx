import { cleanup, render } from '@testing-library/react'

import FilmPosterSection from './film-poster-section'

import { Film } from '@/types/film'

const props: Film[] = [
  {
    id: 'abcdefg',
    title: 'New Hope',
    episodeID: 1
  },
  {
    id: 'efghijk',
    title: 'Empire Strike Back',
    episodeID: 2
  }
]

const mockPersonCard = jest.fn()
jest.mock('@/molecules/film-card', () => () => {
  mockPersonCard(props)
  return <div>film card</div>
})

afterEach(() => {
  cleanup
  jest.clearAllMocks()
})

describe('Film Poster Section', () => {
  it('renders without crashing', () => {
    render(<FilmPosterSection films={props} />)
  })

  it('render title', () => {
    const { getByTestId } = render(<FilmPosterSection films={props} />)
    expect(getByTestId('title')).toHaveTextContent('Movie')
  })

  it('render film card as many as props', () => {
    render(<FilmPosterSection films={props} />)
    expect(mockPersonCard).toHaveBeenCalledTimes(2)
    expect(mockPersonCard).toHaveBeenCalledWith(props)
  })

  it('do not render film card when film props in empty or undefined', () => {
    render(<FilmPosterSection films={[]} />)
    expect(mockPersonCard).not.toHaveBeenCalled()
  })

  it('matches snapshot', () => {
    const tree = render(<FilmPosterSection films={props} />)
    expect(tree.container).toMatchSnapshot()
  })
})
