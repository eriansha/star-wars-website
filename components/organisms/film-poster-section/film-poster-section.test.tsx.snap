// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Poster Section matches snapshot 1`] = `
<div>
  <section
    class="bg-gray-dark pb-24 pt-10 md:pt-24"
  >
    <div
      class="flex flex-col items-center"
    >
      <h1
        class="font-bold pb-16 text-5xl"
        data-testid="title"
      >
        Movie
      </h1>
      <div
        class="flex flex-wrap gap-10 items-center justify-evenly mx-8"
      >
        <div>
          film card
        </div>
        <div>
          film card
        </div>
      </div>
    </div>
  </section>
</div>
`;
